core/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -68,9 +64,7 @@ public IngestDocument(String index, String type, String id, String routing, Stri
         }
 
         this.ingestMetadata = new HashMap<>();
-        DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZZ", Locale.ROOT);
-        df.setTimeZone(TimeZone.getTimeZone("UTC"));
-        this.ingestMetadata.put(TIMESTAMP, df.format(new Date()));
+        this.ingestMetadata.put(TIMESTAMP, new Date());
```

Comment:
Strictly speaking this isn't a breaking change.

In `PipelineExecutionService#innerExecute(...)` when we serialize the map, the `Date` instance will be serialized by `XContentBuilder#DEFAULT_DATE_PRINTER` formatter, which uses this format `yyyy-MM-dd'T'HH:mm:ss.SSSZZ`, so things remain to be the same.